src/viam/ur/module/ur_arm.cpp

```diff
@@ -215,6 +226,20 @@ URArm::URArm(Model model, const Dependencies& deps, const ResourceConfig& cfg) :
         throw std::runtime_error("couldn't connect to dashboard");
     }
 
+    // verify that the connected arm is the same model as the configured module
+    {
+        std::string conn_model_type{};
+        if (!current_state_->dashboard->commandGetRobotModel(conn_model_type)) {
+            throw std::runtime_error("failed to get model info of connected arm");
+        }
+
+        std::string cfg_model_type = to_robot_type_string(model_.to_string());
+        if (cfg_model_type != conn_model_type) {
+            const auto err = "configured model type `" + cfg_model_type + "` does not match connected arm `" + conn_model_type + "`";
```

Comment:
verified that this error shows up on the component card's error tab.

tried with separate error/debug logs, but it wasn't as easy to find in the noise of the other logs

Reply:
A better way to build up a string is with an `ostringstream`. See https://github.com/viam-modules/viam-mlmodelservice-triton/blob/3026fade112f2406d5b82fab3a570db9b8a8648e/src/viam_mlmodelservice_triton_impl.cpp#L749-L753 for instance